### PR TITLE
transfer-contract: Fix tree-filter for transparent notes with 0 value

### PR DESCRIPTION
**Pull request by moCello**
Moves the filter for transparent notes with value 0 to `tree.push`

### CODE REVIEW

File: contracts/transfer/src/tree.rs [hunk 1]
```diff
@@ -26,7 +28,14 @@ impl Tree {
         }
     }
 
+    /// Push one [`NoteLeaf`] onto the [`Tree`], filtering out notes that are
+    /// transparent and have a value of 0.
     pub fn push(&mut self, mut leaf: NoteLeaf) -> Note {
+        // skip transparent notes with a value of 0
+        if leaf.note.value(None).is_ok_and(|value| value == 0) {
+            return leaf.note;
+        }
```
Comment (herr-seppia) [inline contracts/transfer/src/tree.rs:37]: Should this return an Option?

**Review by herr-seppia — CHANGES_REQUESTED** [review]
LGTM overall
just some question to address:

1) Should we avoid to include empty nodes in the events?
2) is ` transitory::push_note(note);` obsoleted? (refund actually takes the outputs from the tx.outputs instead of from transitory

File: contracts/transfer/src/tree.rs (hunk 1 above)
Comment (moCello) [reply contracts/transfer/src/tree.rs:37]: I had a similar idea but didn't want to change the contract too much

**Review by herr-seppia — APPROVED** [review]
LGTM

**Pull request merged**